src/dmd/mars.d

```diff
@@ -2306,7 +2306,11 @@ bool parseCommandLine(const ref Strings arguments, const size_t argc, ref Param
                 goto Lerror;
         }
         else if (arg == "-unittest")
-            params.useUnitTests = true;
```

Comment:
@alexandrumc : Here is your bug.
Either set `useUnitTests` whenever one of the three options is used, or completely eliminate it in favor of an enum. But you are removing the only place that sets it to `true` but still leaving conditions using it, that's bound to create bugs.

Reply:
Thanks!